### PR TITLE
Add docker installation from repo for debian backport images

### PR DESCRIPTION
**Pull request by dsrivastavv**

### CODE REVIEW

File: lisa/tools/docker.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def run_container(
         docker_run_output: str,
     ) -> None:
         self.run(
-            f"run --name {container_name} " f"{image_name} 1> {docker_run_output} 2>&1",
+            f"run --name {container_name} {image_name} 1> {docker_run_output} 2>&1",
```
Comment (LiliDeng) [inline lisa/tools/docker.py]: is it "1>" is covered in "2>&1" ?

Comment (dsrivastavv) [reply lisa/tools/docker.py]: This was just a formatting change automatically done by black. But I can take a look

File: lisa/tools/docker.py [hunk 2]
```diff
@@ -65,7 +65,14 @@ def start(self, name: str = "docker") -> None:
...
         if isinstance(self.node.os, Debian):
-            self.node.os.install_packages("docker.io")
+            try:
+                self.node.os.install_packages("docker.io")
+            except Exception as e:
+                self._log.error(
+                    f"Failed to install docker.io: {e}, trying to install docker from"
+                    " repo"
+                )
+                self._install_from_repo()
```
Comment (squirrelsc) [inline lisa/tools/docker.py:75]: How about call this method when it's Debian. and install packages for other distros? That will be more clear.

File: lisa/tools/docker.py [hunk 3]
```diff
@@ -65,7 +65,14 @@ def start(self, name: str = "docker") -> None:
     def _install(self) -> bool:
         service_name = "docker"
         if isinstance(self.node.os, Debian):
-            self.node.os.install_packages("docker.io")
+            try:
```
Comment (LiliDeng) [inline lisa/tools/docker.py:68]: ubuntu distro also use the same logic, right?

File: lisa/tools/docker.py (hunk 2 above)
Comment (dsrivastavv) [reply lisa/tools/docker.py:75]: This is not the recommended installation method. Most of the Debian distribution install using `docker.io` package.

File: lisa/tools/docker.py (hunk 3 above)
Comment (dsrivastavv) [reply lisa/tools/docker.py:68]: Most ubuntu versions will work with "docker.io" package installation

File: lisa/tools/docker.py (hunk 1 above)
Comment (dsrivastavv) [reply lisa/tools/docker.py]: I think `run --name {container_name} {image_name} > {docker_run_output} 2>&1` should be enough

**Review by squirrelsc — DISMISSED** [review]

File: lisa/tools/docker.py (hunk 1 above)
Comment (dsrivastavv) [reply lisa/tools/docker.py]: fixed. Above is working fine

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**